src/graph/components/runtime.tsx

```diff
@@ -4,15 +4,40 @@ import {
   addLinkedInteractiveStateListener, IInteractiveStateWithDataset, removeLinkedInteractiveStateListener, IDataset
 } from "@concord-consortium/lara-interactive-api";
 import { Bar } from "react-chartjs-2";
-import { ChartOptions } from "chart.js";
-import { generateChartData, emptyChartData } from "../generate-chart-data";
+import { ChartOptions, LinearScale } from "chart.js";
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import { generateChartData, emptyChartData, CustomChartData } from "../generate-chart-data";
 import css from "./runtime.scss";
 
 export interface IProps {
   authoredState: IAuthoredState;
 }
 
-const getGraphOptions = (authoredState: IAuthoredState, { displayLegend }: { displayLegend: boolean }): ChartOptions => {
+const getGraphOptions = (authoredState: IAuthoredState, chartData?: CustomChartData): ChartOptions => {
+  let yAxisLabel;
+  if (authoredState.useYAxisLabelFromData && chartData && chartData.datasets.length > 0) {
+    yAxisLabel = chartData.datasets[0].label;
+  } else if (!authoredState.useYAxisLabelFromData && authoredState.yAxisLabel) {
+    yAxisLabel = authoredState.yAxisLabel;
+  }
+
+  const yAxis: LinearScale = {
+    scaleLabel: {
+      display: !!yAxisLabel,
+      labelString: yAxisLabel,
+    },
+    ticks: {
+      beginAtZero: true,
+      maxTicksLimit: 5
+    }
+  };
+  if (authoredState.autoscaleYAxis === false) {
+    const maxY = typeof authoredState.yAxisMax === "number" ? authoredState.yAxisMax : 100;
+    // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+    yAxis.ticks!.max = maxY;
```

Comment:
I think `!` and eslint-disable could be avoided by not providing `LinearScale` type in line 24 above.
`yAxis` would be type-checked anyway, lower down, when it's used in line 49. 
But I'm not 100% sure and the current approach is fine too if you don't have time to fiddle with it. 😉